### ISSUE TITLE
How do you 

### ISSUE THREAD

**Issue by JoyHuYY1412** [facebookresearch/maskrcnn-benchmark#619]

**Comment by JoyHuYY1412** [comment]
https://github.com/facebookresearch/maskrcnn-benchmark/blob/05feadff540c0d43e6752db0513d21e41038dbdb/maskrcnn_benchmark/engine/trainer.py#L68-L77

I am a little confused when training with multi-GPU. Does this code mean the losses are accumulated from all GPUs and backward together? But how? since I didn't see any summation here.

**Comment by fmassa** [comment]
This all happens inside `torch.distributed`, thanks to `DistributedDataParallel`.
And note that the code is run with `torch.distributed.launch`.

Check https://pytorch.org/docs/stable/nn.html#distributeddataparallel for more information

**Comment by JoyHuYY1412** [comment]
> This all happens inside `torch.distributed`, thanks to `DistributedDataParallel`.
> And note that the code is run with `torch.distributed.launch`.
> 
> Check https://pytorch.org/docs/stable/nn.html#distributeddataparallel for more information

Thank you for your reply!
BTW, I have another small question with this DistributedDataParallel. For another model, I use the loss in this form:
`loss = max(0 , true_dist + self.alpha - false_dist)  `
like a marginal loss. In case 0 is backwarded, I use if:
```
if loss!=0:
        loss.backward()
        optimizer.step()
```
However, when I use multi-GPU, if loss on one GPU is 0, the code stuck. It doesn't report error but just stopped. I think it may because that no gradient is calculated on one GPU, so it keeps waiting?

I don't know what to do. I really appreciate it if you can help me.

**Comment by chengyangfu** [comment]
Hi @JoyHuYY1412 , 
       This question is discussed in https://github.com/facebookresearch/maskrcnn-benchmark/issues/318 and https://github.com/pytorch/pytorch/issues/13273. You can find some solutions there. 
       In brief, DDP is not well supported right now.
